scripts/linters/general_purpose_linter_test.py

```diff
@@ -326,14 +324,40 @@ def test_linter_with_no_files(self) -> None:
         self.assertFalse(lint_task_report[0].failed)
 
     def test_file_with_no_newline_at_eof(self) -> None:
+        temp_file = tempfile.NamedTemporaryFile(
+            mode='w+', suffix='.py', delete=False)
+
+        content = """from __future__ import annotations
```

Comment:
Note that this only doesn't have a newline at the end of the string "by chance" -- the last line of the file is actually the string "                        " (line 349).

Could you please reformat this similar to what is done in core/controllers/editor.py (around line 370) so that this mimics an actual Python file?

If you like, you can also move this constant up to the top of this file and call it INVALID_NO_NEWLINE_FILE_CONTENT. That might be a little cleaner. You can also leave a content above the `temp_file.write(content)` line below explaining that we use a temporary file here rather than an actual one because the Black formatter auto-fixes newlines at the end of files in LINTER_TESTS_DIR when it is run.

Reply:
Done.